regxmllib/src/main/java/com/sandflow/smpte/regxml/FragmentBuilder.java

```diff
@@ -566,6 +567,8 @@ private void readCharacters(InputStream value, CharacterTypeDefinition definitio
             in = new InputStreamReader(value, "UTF-16BE");
         } else if (definition.getIdentification().equals(Char_UL)) {
             in = new InputStreamReader(value, "US-ASCII");
+        } else if (definition.getIdentification().equals(UTF8Character_UL)) {
+            in = new InputStreamReader(value, "UTF-8");

```

Comment:
Ok. I think we should add the following note:
```
/* NOTE: UTF-8 character encoding is not specified in ST 377-1*/
```
Also, have you checked with @thomasheritage ? I recall this topic of character encoding being discussed at length.